File: app/src/main/java/co/tpcreative/supersafe/model/MainCategoryEntityModel.java
package co.tpcreative.supersafe.model;
import java.io.Serializable;

import co.tpcreative.supersafe.common.entities.MainCategoryEntity;
import co.tpcreative.supersafe.common.util.Utils;

public class MainCategoryEntityModel implements Serializable {
    public int id;
    public String image;
    public String icon;
    public long categories_max;
    public String categories_local_id;
    public String categories_id;
    public String categories_hex_name;
    public String categories_name;
    public boolean isDelete ;
    public boolean isChange ;
    public boolean isSyncOwnServer;
    public boolean isFakePin;
    public String pin ;
    public boolean isCustom_Cover;
    public String items_id;
    public String mainCategories_Local_Id;
    public String unique_id;

    /*Fetch data from local db*/
    public MainCategoryEntityModel(MainCategoryEntity entity){
        this.id = entity.id;
        this.image = entity.image;
        this.icon = entity.icon;
        this.categories_max = entity.categories_max;
        this.categories_local_id = entity.categories_local_id;
        this.categories_id = entity.categories_id;
        this.categories_hex_name = entity.categories_hex_name;
        this.categories_name = entity.categories_name;
        this.isDelete = entity.isDelete;
        this.isChange  = entity.isChange;
        this.isSyncOwnServer = entity.isSyncOwnServer;
        this.isFakePin = entity.isFakePin;
        this.pin  = entity.pin;
        this.isCustom_Cover = entity.isCustom_Cover;
        this.items_id = entity.items_id;
        this.mainCategories_Local_Id = entity.mainCategories_Local_Id;
        this.unique_id = Utils.getUUId();
    }

    /*push data to local db*/
    public MainCategoryEntityModel(MainCategoryModel entity){
        this.id = entity.id;
        this.image = entity.image;
        this.icon = entity.icon;
        this.categories_max = entity.categories_max;
        this.categories_local_id = entity.categories_local_id;
        this.categories_id = entity.categories_id;
        this.categories_hex_name = entity.categories_hex_name;
        this.categories_name = entity.categories_name;
        this.isDelete = entity.isDelete;
        this.isChange  = entity.isChange;
        this.isSyncOwnServer = entity.isSyncOwnServer;
        this.isFakePin = entity.isFakePin;
        this.pin  = entity.pin;
        this.isCustom_Cover = entity.isCustom_Cover;
        this.items_id = entity.items_id;
        this.mainCategories_Local_Id = entity.mainCategories_Local_Id;
        this.unique_id = entity.unique_id;
    }
}

File: app/src/main/java/co/tpcreative/supersafe/model/MainCategoryEntityModel.kt
package co.tpcreative.supersafe.model
import co.tpcreative.supersafe.common.entities.MainCategoryEntity
import co.tpcreative.supersafe.common.util.Utils
import java.io.Serializable

class MainCategoryEntityModel : Serializable {
    var id: Int
    var image: String?
    var icon: String?
    var categories_max: Long
    var categories_local_id: String?
    var categories_id: String?
    var categories_hex_name: String?
    var categories_name: String?
    var isDelete: Boolean
    var isChange: Boolean
    var isSyncOwnServer: Boolean
    var isFakePin: Boolean
    var pin: String?
    var isCustom_Cover: Boolean
    var items_id: String?
    var mainCategories_Local_Id: String?
    var unique_id: String?

    /*Added more fields*/
    var created_date : String? = null
    var updated_date : String? = null
    var date_time  : String? = null

    /*Fetch data from local db*/
    constructor(entity: MainCategoryEntity) {
        id = entity.id
        image = entity.image
        icon = entity.icon
        categories_max = entity.categories_max
        categories_local_id = entity.categories_local_id
        categories_id = entity.categories_id
        categories_hex_name = entity.categories_hex_name
        categories_name = entity.categories_name
        isDelete = entity.isDelete
        isChange = entity.isChange
        isSyncOwnServer = entity.isSyncOwnServer
        isFakePin = entity.isFakePin
        pin = entity.pin
        isCustom_Cover = entity.isCustom_Cover
        items_id = entity.items_id
        mainCategories_Local_Id = entity.mainCategories_Local_Id
        unique_id = Utils.getUUId()
        created_date = entity.created_date
        updated_date = entity.updated_date
        date_time = entity.date_time
    }

    /*push data to local db*/
    constructor(entity: MainCategoryModel) {
        id = entity.id
        image = entity.image
        icon = entity.icon
        categories_max = entity.categories_max
        categories_local_id = entity.categories_local_id
        categories_id = entity.categories_id
        categories_hex_name = entity.categories_hex_name
        categories_name = entity.categories_name
        isDelete = entity.isDelete
        isChange = entity.isChange
        isSyncOwnServer = entity.isSyncOwnServer
        isFakePin = entity.isFakePin
        if (entity.pin == null) {
            entity.pin = ""
        }
        pin = entity.pin
        isCustom_Cover = entity.isCustom_Cover
        items_id = entity.items_id
        mainCategories_Local_Id = entity.mainCategories_Local_Id
        unique_id = entity.unique_id
        created_date = entity.created_date
        updated_date = entity.updated_date
        date_time = entity.date_time
    }
}